fix: Ensuring alphabetical sorting of packages

BalaadityaPatanjali opened the pull request:
Fixes #28 
- Added sorting function in lock.go
- Also updated the test file of lock.go

BalaadityaPatanjali commented:
@dr-housemd Please have a look at the PR and let me know if there are any additional changes or improvements needed.

dr-housemd commented:
Hey @BalaadityaPatanjali  , thanks for contributing! Can you please follow this and make a dedicated commit for adding yourself to CONTRIBUTORS.md?
https://github.com/buildsafedev/bsf?tab=readme-ov-file#contributing

dr-housemd commented on pkg/hcl2nix/lock.go:
```diff
@@ -226,5 +227,23 @@ func mapPackageCategory(packages Packages) map[string][]Category {
+	return sortedMap
+}
+
+func sortMapKeys(m map[string][]Category) []string {
```
I don't think maps ordering is reliable in Go(even if it looks like it works sometimes). It's a victim of the infamous [Hyrum's Law](https://www.hyrumslaw.com/)
  
I think what we should do instead of sort the slices alphabetically every time(once we get a slice back from map, etc).  Does that make sense?

https://groups.google.com/g/golang-nuts/c/TDwGcRQe6mQ?pli=1

BalaadityaPatanjali replied:
Thankyou for the review ! I will try to make the needed changes.

BalaadityaPatanjali commented:
@dr-housemd 
Now, i have copied my keys and values in slices and sorted them. Updated the test function accordingly and it is failing if i change the order of the package. So, i think i have changed it correctly. Please have a look and let me know if there are any additional changes or improvements needed.

dr-housemd commented:
@BalaadityaPatanjali  I think a simpler way would be to just sort this slice here alphabetically. This would ensure that the lock file generated is deterministic.  
https://github.com/buildsafedev/bsf/blob/38b5e573e495b357f470f03e609bceec232bef98/pkg/hcl2nix/lock.go#L142
The other place where we need to sort is here. 
https://github.com/buildsafedev/bsf/blob/38b5e573e495b357f470f03e609bceec232bef98/cmd/update/update.go#L65

PS: can you rebase your branch with main?

BalaadityaPatanjali commented:
> @BalaadityaPatanjali I think a simpler way would be to just sort this slice here alphabetically and not attempt to manipulate maps. This would ensure that the lock file generated is deterministic.
> 
> https://github.com/buildsafedev/bsf/blob/38b5e573e495b357f470f03e609bceec232bef98/pkg/hcl2nix/lock.go#L142
> 
> 
> The other place where we need to sort is here so "bsf update" is determisitic.
> https://github.com/buildsafedev/bsf/blob/38b5e573e495b357f470f03e609bceec232bef98/cmd/update/update.go#L65
> 
> PS: can you rebase your branch with main?

thankyou for the guidance ! i'll start working on it.
